pkg/scheduler/schedule_one.go

```diff
@@ -106,14 +130,57 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 		metrics.Goroutines.WithLabelValues(metrics.Binding).Inc()
 		defer metrics.Goroutines.WithLabelValues(metrics.Binding).Dec()
 
-		sched.bindingCycle(bindingCycleCtx, state, fwk, scheduleResult, assumedPodInfo, podsToActivate, start)
+		status := sched.bindingCycle(bindingCycleCtx, state, fwk, scheduleResult, assumedPodInfo, start)
+		if !status.IsSuccess() {
```

Comment:
Similar to schedulingCycle, moved the `metrics recording`, `unreserve plugins running`, `forgetPod` and `failureHandler` out of bindingCycle.